src/pages/Signup.jsx

```diff
@@ -1,7 +1,204 @@
-import React from "react";
+import axios from "axios";
+import React, { useState } from "react";
+import { useNavigate } from "react-router-dom";
+import { registerUser } from "../state/util/index";
 
 const Signup = () => {
-  return <div>Signup</div>;
+  const navigate = useNavigate();
+  const [signUpForm, updateSignUpForm] = useState({
+    email: "   ",
+    password: "",
+    confirmPassword: "",
+    firstName: "",
+    lastName: "",
+  });
+
+  const [validationMessage, setValidationMessage] = useState({
+    email: "",
+    password: "",
+    confirmPassword: "",
+    firstName: "",
+    lastName: "",
+  });
+
+  const signUpFormSubmission = async () => {
+    if (signUpForm.firstName.trim() === "") {
+      setValidationMessage((prev) => {
+        return { ...prev, firstName: "firstName can not be empty" };
+      });
+    } else {
+      setValidationMessage((prev) => {
+        return { ...prev, firstName: "" };
+      });
+    }
+    if (signUpForm.lastName.trim() === "") {
+      setValidationMessage((prev) => {
+        return { ...prev, lastName: "lastName can not be empty" };
+      });
+    } else {
+      setValidationMessage((prev) => {
+        return { ...prev, lastName: "" };
+      });
+    }
+    if (
+      signUpForm.email.trim() === "" ||
+      !signUpForm.email.includes("@") ||
+      !signUpForm.email.includes(".")
+    ) {
+      setValidationMessage((prev) => {
+        return { ...prev, email: "email must include @ and ." };
+      });
+    } else {
+      setValidationMessage((prev) => {
+        return { ...prev, email: "" };
+      });
+    }
+    if (
+      signUpForm.password.trim() === "" ||
+      !signUpForm.password.includes("@") ||
+      !/\d/.test(signUpForm.password)
+    ) {
+      setValidationMessage((prev) => {
+        return {
+          ...prev,
+          password: "email must include @ and one numberic char",
+        };
+      });
+      console.log("email");
+    } else {
+      setValidationMessage((prev) => {
+        return { ...prev, password: "" };
+      });
+    }
+
+    if (signUpForm.confirmPassword !== signUpForm.password) {
+      setValidationMessage((prev) => {
+        return {
+          ...prev,
+          confirmPassword: "please put the same password",
+        };
+      });
+    } else {
+      setValidationMessage((prev) => {
+        return {
+          ...prev,
+          confirmPassword: "",
+        };
+      });
+    }
+
+    console.log(validationMessage.password, "validateionsfsfsfs");
+    if (
+      !validationMessage.firstName &&
```

Comment:
all these values can be destructured for better readability

Reply:
will refactor this in my next pr